test/test_cpp_api_parity.py

```diff
@@ -1,687 +1,54 @@
-            "or set `module_name` using format `ModuleName`."
-
-        module_full_name = 'torch.nn.' + module_name
-        if module_full_name not in parity_table['torch.nn']:
-            raise RuntimeError(
-                'Module `{}` is not found in Python / C++ API parity table. Please update parity table at {}.'.format(
-                    module_full_name, parity_table_path))
+PARITY_TABLE_PATH = os.path.join(os.path.dirname(__file__), 'cpp_api_parity', 'parity-tracker.md')
 
-        has_impl_parity, _ = parity_table['torch.nn'][module_full_name]
+parity_table = parse_parity_tracker_table(PARITY_TABLE_PATH)
 
-        def add_ctor_args_test_for_module(module_name, has_impl_parity):
-            ctor_args_test_name = 'test_torch_nn_{}_ctor_args'.format(module_name)
-
-            def ctor_args_test(self):
-                self._test_torch_nn_module_ctor_args(
-                    module_name=self._testMethodName.replace('test_torch_nn_', '').replace('_ctor_args', ''))
-
-            if not has_impl_parity:
-                ctor_args_test = unittest.expectedFailure(ctor_args_test)
-
-            # We only run one constructor args test per module
-            if not has_test(ctor_args_test_name):
-                add_test(ctor_args_test_name, ctor_args_test)
-
-        def add_variant_test_for_module(module_name, test_params_dict, has_impl_parity):
-            module_metadata = torch_nn_modules.module_metadata_map[module_name]
-            for device in devices:
-                test_params = _process_test_params(
-                    test_params_dict=test_params_dict,
-                    module_metadata=module_metadata,
-                    device=device,
-                    is_criterion=is_criterion)
-                test_name = 'test_torch_nn_{}'.format(test_params.module_variant_name)
-                torch_nn_test_params_map[test_name] = test_params
-
-                def test_fn(self):
-                    self._test_torch_nn_module_variant(test_params=torch_nn_test_params_map[self._testMethodName])
-
-                if device == 'cuda':
-                    test_fn = unittest.skipIf(not TEST_CUDA, "CUDA unavailable")(test_fn)
-
-                if not has_impl_parity:
-                    test_fn = unittest.expectedFailure(test_fn)
-
-                add_test(test_name, test_fn)
-
-        add_ctor_args_test_for_module(module_name, has_impl_parity)
-        add_variant_test_for_module(module_name, test_params_dict, has_impl_parity)
-
-add_torch_nn_module_tests(
-    sample_module.module_tests + common_nn.module_tests + common_nn.new_module_tests,
-    is_criterion=False)
-
-add_torch_nn_module_tests(
-    common_nn.criterion_tests + common_nn.new_criterion_tests,
-    is_criterion=True)
-
-# Assert that there exists auto-generated tests for SampleModule.
+class TestCppApiParity(common.TestCase):
+    pass
+
+expected_test_params_dicts = []
+
+for test_params_dicts, test_instance_class in [
```

Comment:
I think the following logic will make it more explicit that we're not accidentally duplicating the tests. I know that we have some asserts over the place to check this, but it's not entirely obvious from a developer's perspective because those asserts are elsewhere and not here
```
for test_params_dict, test_instance_class in [
    ...
]:
    for test_params_dict in test_params_dict:
        if is_functional_test(test_params_dict):
            functional_impl_check.write_test(TestCppApiParity, test_param_dict, test_instance_class, parity_table, devices)
        else:
             module_impl_check.write_test(TestCppApiParity, ...)
```

Reply:
Oh, the assert is actually on the next line lol.